infrastructure/channels/serialize_child.html

```diff
@@ -0,0 +1,167 @@
+<!doctype html>
+<script src="/resources/channel.sub.js"></script>
+<script src="serialize-data.js"></script>
+<script>
+
+let lastData;
+
+// Hack: these will be recreated as assert failures in the caller
+function AssertionError(message) {
+    this.message = message;
+}
+AssertionError.prototype = Object.create(Error.prototype);
+
+function compareResult(name, actual) {
+    let obj = objects[name]
+    let expected = obj.hasOwnProperty("output") ? obj.output : obj.input;
+    seen = new Set();
+    try {
+        compareValue(actual, expected, seen);
+    } catch(e) {
+        throw new AssertionError(e.message);
+    }
+    return true;
+}
+
+function compareValue(actualValue, expectedValue, seen) {
+    if (typeof actualValue != typeof expectedValue) {
+        throw new Error(`Types differ, expected ${typeof expectedValue}, got ${typeof actualValue}`);
+    }
+    if (["undefined", "string", "boolean", "number", "bigint"].includes(typeof expectedValue) ||
+        actualValue === null) {
+        if (Number.isNaN(expectedValue)) {
+            if (!Number.isNaN(actualValue)) {
+                throw new Error("Expected NaN, got $(actual)");
+            }
+        } else if (actualValue !== expectedValue) {
+            throw new Error(`Expected ${typeof expected} ${expected}, got ${actual}`);
+        }
+        return;
+    }
+
+    if (expectedValue.constructor && actualValue.constructor && expectedValue.constructor.name !== actualValue.constructor.name) {
+        throw new Error(`Constructors differ, expected ${expectedValue.constructor.name}, got ${actualValue.constructor.name}`);
+    }
+    if (expectedValue.constructor.name === "SendChannel") {
```

Comment:
Going just by the local code, `expectedValue.constructor` could still be undefined here. Otherwise, why bother checking it above?